crates/pink/pink-extension/Cargo.toml

```diff
@@ -7,11 +7,12 @@ license = "Apache-2.0"
 keywords = ["fat-contract", "pink", "ink"]
 
 [dependencies]
-ink = { version = "4", default-features = false, features = ["ink-debug"] }
+ink = { version = "4.0.1", default-features = false, features = ["ink-debug"] }
 scale = { package = "parity-scale-codec", version = "3.1", default-features = false, features = ["derive"] }
 scale-info = { version = "2.1", default-features = false, features = ["derive"], optional = true }
 pink-extension-macro = { version = "0.4.0", path = "./macro" }
 log = "0.4.17"
+dlmalloc = { version = "0.2.4", default-features = false, features = ["global"], optional = true }
```

Comment:
Should we enable `dlmalloc` feature in all the pink contracts?

Reply:
Some contracts might want to use a custom allocator though. The phat-quickjs, for example, would be better to use the allocator from the C side. In qjs, we currently delegate the C-side `malloc`/`free` to Rust-side doesn't feel so good.